Cython/Compiler/Optimize.py

```diff
@@ -2580,26 +2588,53 @@ def _handle_simple_function_set(self, node, function, pos_args):
                 is_temp=node.is_temp,
                 py_name="set"))
 
-    PyFrozenSet_New_func_type = PyrexTypes.CFuncType(
-        Builtin.frozenset_type, [
-            PyrexTypes.CFuncTypeArg("it", PyrexTypes.py_object_type, None)
-        ])
-
+    # When calling frozenset(), we replace SimpleCallNode
     def _handle_simple_function_frozenset(self, node, function, pos_args):
+        # empty frozenset is singleton, so we never try to free them
         if not pos_args:
-            pos_args = [ExprNodes.NullNode(node.pos)]
-        elif len(pos_args) > 1:
+            result = ExprNodes.FrozenSetNode(node.pos, is_temp=False, args=None)
+            result.is_literal = True
+            return result
+
+        # frozenset() expects no arguments, or one iterable argument
+        # https://docs.python.org/3.9/library/stdtypes.html#frozenset
```

Comment:
This seems a low-information comment. People who look at this method probably already know how `frozenset()` is used.